clang/include/clang/Basic/Module.h

```diff
@@ -189,6 +189,10 @@ class alignas(8) Module {
   /// The \c ActionCache key for this module, if any.
   Optional<std::string> ModuleCacheKey;
 
+  /// The CAS filesystem root ID for implicit modules built with the dependency
+  /// scanner, if any.
+  Optional<std::string> CASFileSystemRootID;
```

Comment:
We could introduce `CASManager` and stash CAS-related state in there.

PCM files contain information that's tied to the module but end up being deserialized into other objects (`ASTContext`, `Preprocessor`, `FileManager`, `SourceManager` etc.).

Reply:
Summary of offline discussion:
* Ben would prefer not to have a CASManager-type object, because there isn't really a single "thing" that is being managed - the CAS has different uses in different places, so this would be a grab bag of state.
* We don't like having build/serialization related state in Module itself. That includes these cas fields but also the existing ASTFile and ASTFileSignature.
* We liked the idea of a ModuleBuildManager that would extract specifically these Module fields.  Ben tried to make this refactoring, but it turns out to be fairly messy due to how these fields are used throughout the compiler -- Examples
  * libclang API that assume Module can retrieve ASTFile, which would necessitate a new fat object be allocated an managed on the CXTU, 
  * CodeGen is designed to be an ASTConsumer rather than having a lot of direct access to CompilerInstance objects, but it requires the ASTFile for debug info.
* For now, I would like to leave the cas-related fields here. If we ever do succeed in factoring out ASTFile I think that would be the right time to move these as well.